services/QuillLMS/spec/controllers/feedback_history_ratings_controller_spec.rb

```diff
@@ -85,4 +85,30 @@
 
   end
 
+  describe "PUT #mass_mark" do
+    it "should create or update all relevant records" do
```

Comment:
Not critical, but it may be worth adding an `it` case for invalid records just to establish how we expect them to be handled (all valid get saved, but invalids return errors; nothing gets saved if any are invalid; whatever...)

Reply:
So, after about an hour of testing, I can't find a way to actually submit this request with data that will create an invalid record. There may be a way to stub an error for the sake of the test, but since the actual existence of an error is theoretical I think I'm going to punt on it for now if that's okay with you.

Reply:
So I think that if you passed `nil` in as the value for the rating, you'd violate a database constraint, and that could happen in the controller since we're using `permit` for `rating` instead of `require`.  That said, I think that you're point here is valid.  We don't expect invalid values here, so we don't have to create a test case to deal with them.

So let's just try to solve this by replacing the `permit` with `require` in the `feedback_history_rating_params` since that should prevent `nil` values from sneaking in there.  Also, let's replace all instances of `params` calls in the controller with `feedback_history_rating_params` since that'll run that validation for us.

Reply:
Per discussion, adding a new private method for the `mass_mark` params since they are slightly different than the `create_or_update` params.